daemon/logger/local/local_test.go

```diff
@@ -188,16 +189,39 @@ func BenchmarkLogWrite(b *testing.B) {
 		b.Run(fmt.Sprintf("%d", len(data)), func(b *testing.B) {
 			entry := &logdriver.LogEntry{Line: data, Source: "stdout", TimeNano: t.UnixNano()}
 			b.SetBytes(int64(entry.Size() + encodeBinaryLen + encodeBinaryLen))
-			b.ResetTimer()
-			for i := 0; i < b.N; i++ {
-				msg := logger.NewMessage()
-				msg.Line = data
-				msg.Timestamp = t
-				msg.Source = "stdout"
-				if err := local.Log(msg); err != nil {
-					b.Fatal(err)
+
+			var wg sync.WaitGroup
```

Comment:
What are we gaining by changing this benchmark?

Note that this doesn't run in CI.

Reply:
We are gaining verification that there was a race condition in WriteLogEntry and that it is now fixed, as I detailed in the PR description and commit message.

I am aware that benchmarks are not run in CI. Unit tests are not run in CI with the race detector enabled, either. (If they were, the `daemon/logger/loggerutils/cache` tests would consistently fail.) So either way someone needs to run `go test -race` manually to verify there are no data races.

Reply:
I think I'd prefer to not change a benchmark.